simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/utils/Cache.kt

```diff
@@ -0,0 +1,166 @@
+package org.nypl.simplified.cardcreator.utils
+
+import android.content.SharedPreferences
+import androidx.core.content.edit
+import org.nypl.simplified.cardcreator.models.AccountInformation
+import org.nypl.simplified.cardcreator.models.AddressDetails
+import org.nypl.simplified.cardcreator.models.PersonalInformation
+
+/**
+ * Util class for handling saving/retrieving address across card creator process
+ * This class acts as private key value pair caching system
+ */
+class Cache internal constructor(private val sharedPreferences: SharedPreferences) {
+
```

Comment:
I'd suggest adding a secondary constructor here that passes in a default SharedPreferences instance:

```kotlin
class Cache internal constructor(
  private val preferences: SharedPreferences
) {
  constructor(context: Context) : this(
    context.getSharedPreferences("CARD_CREATOR", Context.MODE_PRIVATE)
  )
}
```

This should eliminate a bunch of boilerplate code in your fragments.